libraries/eosiolib/core/eosio/crypto_ext.hpp

```diff
@@ -48,14 +49,183 @@ namespace eosio {
          return dg;
       }
    }
-      
+
    /**
     *  @defgroup crypto Crypto
     *  @ingroup core
     *  @brief Defines API for calculating and checking hashes which
     *  require activating crypto protocol feature
     */
 
+   /**
+    * Abstracts G1 and G2 points. Ensures sizes of x and y are valid
+    *
+    *  @ingroup crypto
+    */
+   struct point_view {
+      /**
+       * Pointer to the x coordinate
+       */
+      char* x;
+
+      /**
+       * Pointer to the y coordinate
+       */
+      char* y;
+
+      /**
+       * Number of bytes in each of x and y
+       */
+      uint32_t size;
+
+      /**
+       * Construct a point given x and y
+       *
+       * @param x_ - The x coordinate, a vector of chars
+       * @param y_ - The y coordinate, a vector of chars
+       */
+      point_view(std::vector<char>& x_, std::vector<char>& y_)
+      :x(x_.data()), y(y_.data()), size(x_.size())
+      {
+         eosio::check( x_.size() == y_.size(), "x's size must be equal to y's" );
+      };
+
+      /**
+       * Construct a point given a serialized point
+       *
+       * @param p - The serialized point
+       */
+      point_view(std::vector<char>& p)
+      :x(p.data()), y(p.data() + p.size()/2), size(p.size()/2)
+      {
+      };
+
+      /**
+       *  Return serialzed point containing only x and y
+       */
+      std::vector<char> serialized() const {
+         std::vector<char> x_and_y( x, x + size );
+         x_and_y.insert( x_and_y.end(), y, y + size );
+         return x_and_y;
+      }
+
+      /**
+      * Copy a serialized buffer to current point
+      *
+      * @param source - The source to be copied from
+      */
+      void copy_from(const std::vector<char>& source) {
+         eosio::check( source.size() == 2 * size, "size of souce buffer must be size of x + y" );
+         std::memcpy(x, source.data(), size);
+         std::memcpy(y, source.data() + size, size);
+      }
+   };
+
+   static constexpr size_t g1_coordinate_size = 32;
+   static constexpr size_t g2_coordinate_size = 64;
+   static constexpr size_t blake2f_result_size = 64;
+
+   /**
+    * Abstracts G1 point. Ensures sizes of x and y are 32
+    *
+    *  @ingroup crypto
+    */
+   struct g1_view : public point_view {
```

Comment:
I would just make these templated and add a typedefs.

So, something like 
```c++
template <std::size_t Size = 32>
struct point_view {
   point_view(const std::vector<char>& x, const std::vector<char>& y) {
      >> The other constructor code here from above <<
      eosio::check ( size == Size, "point size must match");
   }
   
   .
   .
   .

   using g1_view = point_view<g1_coordinate_size>;
   using g2_view = point_view<g2_coordinate_size>;

Reply:
Will change. Thanks.